internal/plugin/metrics_plugin.go

```diff
@@ -26,60 +26,60 @@ const (
 )
 
 type (
-	shutdownFunc func()
-
 	// The Metrics plugin. Discovers and owns the data source that produce metrics for the Agent.
 	Metrics struct {
 		producers       map[model.MetricsSourceType]model.MetricsProducer
 		exporters       map[model.ExporterType]model.Exporter
-		conf            config.Config
+		config          *config.Config
 		pipe            bus.MessagePipeInterface
+		shutdownFuncs   []*context.CancelFunc
```

Comment:
nit: `context.CancelFunc` is a function reference, so I don't understand why it would need to be a pointer?
```suggestion
		shutdownFuncs   []context.CancelFunc
```